src/mechanics/camera.rs

```diff
@@ -7,7 +7,8 @@ pub fn move_camera(
 ) {
     if let Ok(player_transform) = player_query.get_single() {
         if let Ok(mut camera_transform) = camera_query.get_single_mut() {
-            camera_transform.translation = player_transform.translation;
+            let new_camera_position = Vec3{x:player_transform.translation.x, y:player_transform.translation.y, z:camera_transform.translation.z};
```

Comment:
There's a constructor function called new that takes in three arguments, rather than having to fill the Vec3 yourself. What do you think about using that over initializing the struct yourself?
```rust
let new_camera_position = Vec3::new(player_transform.translation.x, player_transform.translation.y, camera_transform.translation.z);

```